OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -344,7 +344,13 @@ private static void HandleSuffixPronoun(ref StringBuilder formattedString, ReadO
                         continue;
                     case StringFormatEncoder.FormatSuffix.OrdinalIndicator:
                         // TODO: if the preceding expression value is not a float, it should be replaced with 0 (0th)
-                        if (interps[prevInterpIndex].TryGetValueAsFloat(out var ordinalNumber)) {
+                        if (interps[prevInterpIndex].TryGetValueAsInteger(out var ordinalNumber)) {
+
+                            /// For some mystical reason byond converts \th to integers
```

Comment:
would it be better to do `if (interps[prevInterpIndex].TryGetValueAsFloat(out var ordinalNumber))` and cast it to an int?